js/renovation/ui/scroll_view.tsx

```diff
@@ -4,34 +4,173 @@ import {
   JSXComponent,
   Slot,
   OneWay,
+  Method,
+  Ref,
 } from 'devextreme-generator/component_declaration/common';
+import { isNumeric } from '../../core/utils/type';
+import { Widget } from './common/widget';
+import config from '../../core/config';
 
-export const viewFunction = (viewModel: ScrollView) => (
-  <div className={viewModel.cssClasses}>
+const DIRECTION_VERTICAL = 'vertical';
+const DIRECTION_HORIZONTAL = 'horizontal';
+const DIRECTION_BOTH = 'both';
+const SCROLLABLE_CONTENT_CLASS = 'dx-scrollable-content';
+
+export interface Location {
+  top: number;
+  left: number;
+}
+
+export const ensureLocation = (location: number | Location): Location => {
+  if (isNumeric(location)) {
```

Comment:
May be we should add type.d.ts whith type guard version of isNumeric definition:
declare function isNumeric<T>(possibleNumber: number | T): possibleNumber is number;